### PR TITLE
Added Geotagging

### PR DESCRIPTION
**Pull request by chaviw**

### CODE REVIEW

**Comment by ne0fhyk** [comment]
@chaviw can you bump the dronekit-client library version to `2.7.0`.

File: Android/AndroidManifest.xml [hunk 1]
```diff
@@ -138,19 +139,30 @@
...
+            android:theme="@style/CustomActionBarTheme.Transparent"/>
+
+        <activity
+            android:name=".activities.GeoTagActivity"
+            android:label="@string/geo_tag_label"
+            android:launchMode="singleTop"/>
+
+        <activity
+            android:name=".activities.GetRecapImageActivity"
+            android:label="Get Recap Image"
```
Comment (ne0fhyk) [inline Android/AndroidManifest.xml]: @chaviw extract the label to the string resources.

File: Android/build.gradle [hunk 2]
```diff
@@ -24,7 +24,7 @@ dependencies {
     compile "com.android.support:design:${support_lib_version}"
 
     //DroneKit-Android client library.
-    compile 'com.o3dr.android:dronekit-android:2.6.8-beta2'
+    compile 'com.o3dr.android:dronekit-android:2.6.9'
```
Comment (ne0fhyk) [inline Android/build.gradle]: @chaviw bump the version to `2.7.0`.

File: Android/res/drawable/geotag_button.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:state_pressed="true">
+        <shape android:shape="rectangle">
```
Comment (ne0fhyk) [inline Android/res/drawable/geotag_button.xml]: @chaviw you're repeating the same shape for all three states. Instead you can define the shape once, and set a selector for the `solid` color attribute.

File: Android/res/layout-land/fragment_get_camera_logs.xml [hunk 4]
```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/transparent"
```
Comment (ne0fhyk) [inline Android/res/layout-land/fragment_get_camera_logs.xml]: @chaviw the container activity is not full screen or transparent, as such this fragment layout shouldn't need the `android:fitsSystemWindows` attribute.

File: Android/res/layout-port/fragment_get_camera_logs.xml [hunk 5]
```diff
@@ -0,0 +1,97 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/transparent"
+    android:fitsSystemWindows="true"
```
Comment (ne0fhyk) [inline Android/res/layout-port/fragment_get_camera_logs.xml]: @chaviw the container activity is not full screen or transparent, as such this fragment layout shouldn't need the `android:fitsSystemWindows` attribute.

File: Android/res/layout/fragment_finish_geotagging.xml [hunk 6]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/transparent"
+    android:fitsSystemWindows="true"
```
Comment (ne0fhyk) [inline Android/res/layout/fragment_finish_geotagging.xml]: @chaviw the container activity is not full screen or transparent, as such this fragment layout shouldn't need the `android:fitsSystemWindows` attribute.

File: Android/res/layout/activity_geotag.xml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <include layout="@layout/actionbar_toolbar"/>
+
+    <View
```
Comment (ne0fhyk) [inline Android/res/layout/activity_geotag.xml]: @chaviw what's the purpose of this view since it's hidden by the view below?

File: Android/res/layout/fragment_geotag_images.xml [hunk 8]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/transparent"
+    android:fitsSystemWindows="true"
```
Comment (ne0fhyk) [inline Android/res/layout/fragment_geotag_images.xml]: @chaviw the container activity is not full screen or transparent, as such this fragment layout shouldn't need the `android:fitsSystemWindows` attribute.

File: Android/src/org/droidplanner/android/activities/GeoTagActivity.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+        if (actionBarGeoTag == null) {
+            actionBarGeoTag = new ActionBarGeoTagFragment();
+            fm.beginTransaction().add(toolbarId, actionBarGeoTag).commit();
+        }
+    }
+
+    public void finishedLoadingLogs() {
+        FragmentManager fm = getSupportFragmentManager();
+        Fragment fragment = fm.findFragmentById(R.id.fragment_geotag_layout);
+        if (fragment == null || !(fragment instanceof GeoTagImagesFragment)) {
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/activities/GeoTagActivity.java]: @chaviw you can skip the check for `null`. If `fragment` is `null` then the `instanceof` check will also fail.

File: Android/src/org/droidplanner/android/activities/GeoTagActivity.java [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+            fragment = new GeoTagImagesFragment();
+            fm.beginTransaction().replace(R.id.fragment_geotag_layout, fragment).commit();
+        }
+    }
+
+    public void finishedGeotagging(ArrayList<File> files) {
+        FragmentManager fm = getSupportFragmentManager();
+        Fragment fragment = fm.findFragmentById(R.id.fragment_geotag_layout);
+        if (fragment == null || !(fragment instanceof FinishGeoTagFragment)) {
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/activities/GeoTagActivity.java]: @chaviw same here. you can skip the check for `null`.

File: Android/src/org/droidplanner/android/activities/GeoTagActivity.java [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+            fragment.setArguments(args);
+            fm.beginTransaction().replace(R.id.fragment_geotag_layout, fragment).commit();
+        }
+    }
+
+    public void updateTitle(@StringRes int title) {
+        final int toolbarId = getToolbarId();
+        final FragmentManager fm = getSupportFragmentManager();
+        Fragment actionBarGeoTag = fm.findFragmentById(toolbarId);
+        if (actionBarGeoTag != null && actionBarGeoTag instanceof ActionBarGeoTagFragment) {
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/activities/GeoTagActivity.java]: @chaviw same here. you can skip the check for null.

File: Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+                }
+            }
+        });
+
+        int numFiles = 0;
+        Bundle args = getArguments();
+        if (args != null) {
+            ArrayList<File> files = (ArrayList<File>) args.getSerializable(GeoTagImagesService.EXTRA_GEOTAGGED_FILES);
+            numFiles = files.size();
+            if (files != null && numFiles > 0) {
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java]: @chaviw no point checking for `files` being `null` here since you already call `files.size()` above.

File: Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+                    intent.setDataAndType(uri, "resource/folder");
+                    startActivity(Intent.createChooser(intent, "Open folder"));
+                }
+            }
+        });
+
+        int numFiles = 0;
+        Bundle args = getArguments();
+        if (args != null) {
+            ArrayList<File> files = (ArrayList<File>) args.getSerializable(GeoTagImagesService.EXTRA_GEOTAGGED_FILES);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java]: @chaviw what's the list of files argument for? Can't you just pass the files count and the parent directory instead?

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+                    break;
+            }
+        }
+    };
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+
+        if (!(context instanceof GeoTagActivity)) {
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java]: @chaviw should use `getActivity()` here instead of `context`. There's no guarantee that the given `context` is derived from the parent activity.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java [hunk 15]
```diff
@@ -0,0 +1,191 @@
...
+    public void onDetach() {
+        super.onDetach();
+        activity = null;
+    }
+
+    private void startGeoTagging() {
+        Context context = getContext();
+        Intent intent = new Intent(context, GeoTagImagesService.class);
+        intent.setAction(GeoTagImagesService.ACTION_START_GEOTAGGING);
+        getContext().startService(intent);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java]: @chaviw should ensure `getContext()` is not null, and reuse the `context` variable defined at the beginning of the method.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java [hunk 16]
```diff
@@ -0,0 +1,191 @@
...
+    }
+
+    private void cancelGeoTagging() {
+        Context context = getContext();
+        Intent intent = new Intent(context, GeoTagImagesService.class);
+        intent.setAction(GeoTagImagesService.ACTION_CANCEL_GEOTAGGING);
+        getContext().startService(intent);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java]: @chaviw should ensure `getContext()` is not null, and reuse the `context` variable defined at the beginning of the method.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java [hunk 17]
```diff
@@ -0,0 +1,191 @@
...
+    private void failedLoading(String message) {
+        updateState(STATE_INIT);
+        instructionText.setText(String.format(getString(R.string.failed_geotag), message));
+    }
+
+    private void finishedGeotagging(Intent intent) {
+        boolean success = intent.getBooleanExtra(GeoTagImagesService.EXTRA_SUCCESS, false);
+        if (success) {
+            ArrayList<File> files = (ArrayList<File>) intent.getSerializableExtra(GeoTagImagesService.EXTRA_GEOTAGGED_FILES);
+            activity.finishedGeotagging(files);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java]: @chaviw check for `activity` being null.

File: Android/AndroidManifest.xml (hunk 1 above)
Comment (chaviw) [inline Android/AndroidManifest.xml]: @ne0fhyk Actually this entire activity is removed. I deleted it from the manifest

File: Android/res/layout/activity_geotag.xml (hunk 7 above)
Comment (chaviw) [inline Android/res/layout/activity_geotag.xml]: @ne0fhyk This view creates the black background that's behind the button at the bottom. It looks similar to an action bar at the bottom

File: Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java (hunk 12 above)
Comment (chaviw) [inline Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java]: @ne0fhyk True. Moving the files.size() inside if

File: Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java (hunk 13 above)
Comment (chaviw) [inline Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java]: @ne0fhyk It was doing a bunch of stuff with it before. I guess it's not really needed anymore.

File: Android/src/org/droidplanner/android/activities/GeoTagActivity.java [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    public void finishedGeotagging(ArrayList<File> files) {
+        FragmentManager fm = getSupportFragmentManager();
+        Fragment fragment = fm.findFragmentById(R.id.fragment_geotag_layout);
+        if (!(fragment instanceof FinishGeoTagFragment)) {
+            fragment = new FinishGeoTagFragment();
+            Bundle args = new Bundle();
+            args.putInt(NUM_IMAGE_FILES, files.size());
+            if (files != null && files.size() > 0) {
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/activities/GeoTagActivity.java]: @chaviw no point checking for `files` being `null` here since you already called `files.size()` above.

File: Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+
+        int numFiles = 0;
+        Bundle args = getArguments();
+        if (args != null) {
+            numFiles = args.getInt(GeoTagActivity.NUM_IMAGE_FILES);
+            parent = args.getString(GeoTagActivity.PARENT_DIR);
+        }
+
+        if (parent != null) {
+            filesText.setText(String.format(getString(R.string.photos_geotagged), String.valueOf(numFiles), parent));
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java]: @chaviw use [`getString(...)`](https://developer.android.com/reference/android/content/Context.html#getString(int, java.lang.Object...)) instead of `String.format(...)`.

File: Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+        } else {
+            filesText.setText(getString(R.string.no_files_geotagged));
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/FinishGeoTagFragment.java]: @chaviw here, `filesText` being a `TextView`, you can pass the string resource id directly.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java [hunk 21]
```diff
@@ -0,0 +1,199 @@
...
+        lbm = LocalBroadcastManager.getInstance(context);
+
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_geotag_images, container, false);
+    }
+
+    int currState = STATE_INIT;
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java]: @chaviw how do you restore the `currState` on screen rotation?

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java [hunk 22]
```diff
@@ -0,0 +1,389 @@
...
+                cancelLoadingLogs();
+                break;
+            case ACTION_START_GEOTAGGING:
+                startGeoTagging();
+                break;
+            case ACTION_CANCEL_GEOTAGGING:
+                cancelGeoTagging();
+                break;
+        }
+        return super.onStartCommand(intent, flags, startId);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw this is one of the few cases in the Android sdk where you don't need to call `super.onStartCommand(...)`. Look up the [available return values](https://developer.android.com/reference/android/app/Service.html), and select the most appropriate for the task at hand.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java [hunk 23]
```diff
@@ -0,0 +1,389 @@
...
+
+    private void cancelLoadingLogs() {
+        if (getTlogsTask != null) {
+            getTlogsTask.cancel(true);
+            getTlogsTask = null;
+        }
+    }
+
+    private void startGeoTagging() {
+        File folder = getApplicationContext().getExternalFilesDir(null);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw you should be able to call `getExternalFilesDir(null)` directly.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java [hunk 24]
```diff
@@ -0,0 +1,389 @@
...
+            return;
+        }
+
+        Uri uri = Uri.fromFile(tlogFile);
+        Handler handler = new Handler();
+        TLogParser.getAllEventsAsync(handler, uri, new TLogParserFilter() {
+
+            @Override
+            public boolean includeEvent(TLogParser.Event event) {
+                return 180 == event.getMavLinkMessage().msgid;
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw make `180` into a constant.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java [hunk 25]
```diff
@@ -0,0 +1,389 @@
...
+            @Override
+            public boolean shouldIterate() {
+                return true;
+            }
+
+
+        }, new TLogParserCallback() {
+            @Override
+            public void onResult(List<TLogParser.Event> eventList) {
+                if (eventList.size() < 0) {
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw you can use `isEmpty()` instead. It conveys the intent better.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java [hunk 26]
```diff
@@ -0,0 +1,389 @@
...
+        }
+
+        final ArrayList<File> photoFiles = new ArrayList<>();
+
+        List<File> photos = searchDir(extMount);
+        if (photos != null) {
+            photoFiles.addAll(photos);
+        }
+
+        if (photoFiles.size() == 0) {
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw use `isEmpty()` instead. Conveys the intent better.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java [hunk 27]
```diff
@@ -0,0 +1,389 @@
...
+
+    private void geoTagImages(final Context context, File tlogFile) {
+        final String extMount = getExternalStorage(context);
+        if (extMount == null) {
+            return;
+        }
+
+        final ArrayList<File> photoFiles = new ArrayList<>();
+
+        List<File> photos = searchDir(extMount);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw why have two `list` of file? Just use one.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java [hunk 28]
```diff
@@ -0,0 +1,389 @@
+package org.droidplanner.android.fragments.geotag;
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw add javadocs so it's easy for someone else to extend in the future.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java [hunk 29]
```diff
@@ -0,0 +1,389 @@
...
+        if (hasNullFile) {
+            sendFailedGeotaggingIntent("No external storage device found.");
+        } else {
+            sendFailedGeotaggingIntent("Incompatible device. No external SD card reader found.");
+        }
+        return null;
+    }
+
+    private static String findRootPath(File extFile) {
+        File currPath = null;
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw check for `extFile` being `null`.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java [hunk 30]
```diff
@@ -0,0 +1,389 @@
...
+            } catch (IllegalArgumentException e) {
+                //swallow
+                return currPath.getAbsolutePath();
+            }
+        }
+        return currPath.getAbsolutePath();
+    }
+
+    private static List<File> searchDir(String mount) {
+        File photoDir = new File(mount + "/DCIM");
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw make `/DCIM` into a constant.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java [hunk 31]
```diff
@@ -0,0 +1,389 @@
...
+        }
+
+        /**
+         * After completing background task Dismiss the progress dialog
+         */
+        protected void onPostExecute(Boolean result) {
+            Intent intent = new Intent(STATE_FINISHED_LOADING_LOGS);
+            intent.putExtra(EXTRA_SUCCESS, result);
+            lbm.sendBroadcast(intent);
+        }
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw dismiss the progress dialog if the task is cancelled as well.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java [hunk 32]
```diff
@@ -0,0 +1,389 @@
...
+
+import timber.log.Timber;
+
+/**
+ * Created by chavi on 10/19/15.
+ */
+public class GeoTagImagesService extends Service {
+    private static final String SSH_USERNAME = "root";
+    private static final String SSH_PASSWORD = "TjSDBkAu";
+    private static final SshConnection soloSshLink = new SshConnection(BuildConfig.SOLO_LINK_IP, SSH_USERNAME, SSH_PASSWORD);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw this ssh link won't work if the device has a cellular connection. Take a look at the implementation on Solo to see how to get around that.

File: Android/src/org/droidplanner/android/fragments/geotag/GetCameraLogsFragment.java [hunk 33]
```diff
@@ -0,0 +1,236 @@
...
+    private GeoTagActivity activity;
+
+    private TextView instructionText;
+    private TextView secondaryInstruction;
+    private Button geotagButton;
+
+    private ImageView phoneControlDots;
+    private ImageView controlCopterDots;
+
+    private int currState = STATE_INIT;
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GetCameraLogsFragment.java]: @chaviw is `currState` restored on lifecycle event?

File: Android/src/org/droidplanner/android/maps/providers/google_map/GoogleMapPrefFragment.kt [hunk 34]
```diff
@@ -137,6 +141,34 @@ public class GoogleMapPrefFragment : MapProviderPreferences(), EditInputDialog.L
                 sharedPref.edit().putString(PREF_MAPBOX_ACCESS_TOKEN, mapboxToken).apply()
             }
         }
+
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/maps/providers/google_map/GoogleMapPrefFragment.kt]: @chaviw why was this added here?

**Comment by ne0fhyk** [comment]
@chaviw once this pr is in, can you update the wiki with simple instructions to use the geotagging feature.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java (hunk 22 above)
Comment (chaviw) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @ne0fhyk True, but the super does handle START_STICKY and START_STICKY_COMPATIBILITY which is what I want. I can return START_STICKY explicitly if you want

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java (hunk 27 above)
Comment (chaviw) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @ne0fhyk I think it was left over from before where I was looking through multiple directories. I'll remove it

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java (hunk 29 above)
Comment (chaviw) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @ne0fhyk Technically it's being checked before this method is called in getExternalStorage(). I check to see this device even has external storage. I can check here too if you want.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java (hunk 31 above)
Comment (chaviw) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @ne0fhyk What progress dialog are you talking about

File: Android/src/org/droidplanner/android/maps/providers/google_map/GoogleMapPrefFragment.kt (hunk 34 above)
Comment (chaviw) [inline Android/src/org/droidplanner/android/maps/providers/google_map/GoogleMapPrefFragment.kt]: @ne0fhyk Probably didn't mean to

File: Android/build.gradle [hunk 35]
```diff
@@ -182,3 +186,7 @@ def getAppKeyPassword() {
 def getAppKey() {
     return hasProperty('ORG_DROIDPLANNER_ANDROID_KEY_ALIAS') ? ORG_DROIDPLANNER_ANDROID_KEY_ALIAS : ''
 }
+
+def getSoloLinkIp(){
+    return hasProperty('SOLO_LINK_IP') ? SOLO_LINK_IP : ''
```
Comment (ne0fhyk) [inline Android/build.gradle]: @chaviw the build is failing on `travis` because of this line. The empty string should be `'\"\"'` instead.

**Comment by ne0fhyk** [comment]
@chaviw seems like you're still working on this pr. Can you ping me once it's complete.

File: Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java (hunk 31 above)
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java]: @chaviw nevermind. I looked at the javadocs comment without looking at the code.
